File: Python/Trik.py
# Trik https://open.kattis.com/problems/trik

data = list(str(input()))
c = 1
for i in range(len(data)):
	if (data[i] == 'A' and c == 1):
		c = 2
	elif (data[i] == 'A' and c == 2):
		c = 1
	elif (data[i] == 'B' and c == 2):
		c = 3
	elif (data[i] == 'B' and c == 3):
		c = 2
	elif (data[i] == 'C' and c == 3):
		c = 1
	elif (data[i] == 'C' and c == 1):
		c = 3
print(c)

File: Java/Trik.java
// https://open.kattis.com/problems/trik

import java.util.Scanner;

public class Trik {
    public static void main(String args[]) {
        Scanner scanner = new Scanner(System.in);
        int c = 1;
        String s = scanner.nextLine();
        char data[] = s.toCharArray();

        for (int i = 0; i < data.length; i++) {
            if (data[i] == 'A' && c == 1) {
                c = 2;
            } else if (data[i] == 'A' && c == 2) {
                c = 1;
            } else if (data[i] == 'B' && c == 2) {
                c = 3;
            } else if (data[i] == 'B' && c == 3) {
                c = 2;
            } else if (data[i] == 'C' && c == 3) {
                c = 1;
            } else if (data[i] == 'C' && c == 1) {
                c = 3;
            }
        }
        System.out.println(c);
    }
}
